coral/src/app/features/approvals/components/RequestDetailsModal.test.tsx

```diff
@@ -0,0 +1,74 @@
+import { cleanup, render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import RequestDetailsModal from "src/app/features/approvals/components/RequestDetailsModal";
+
+const baseProps = {
+  onClose: jest.fn(),
+  onApprove: jest.fn(),
+  onReject: jest.fn(),
+};
+
+describe.only("RequestDetailsModal.test", () => {
+  afterEach(cleanup);
+
+  it("renders a Modal with correct elements (isLoading false)", () => {
```

Comment:
I would define that as a describe block, render it `beforeAll` and to the assertions in different `it` blocks. That is more readable and will give a better debugging experience if one of the assertions fails.

Reply:
But then it's one `describe` with one `beforeAll` block for each test case. Because we need to pass a different value for `isLoading` (or `disabled`) in the case of the other modal). Seems overkill, and not that much more legible? I would have thought that having the component explicitly rendered in the scope of an `it` block would make it easier?

Reply:
In `renders a Modal with correct elements (isLoading false)` are 8 assertions (or 5 because `toBeVisible()` could be removed). 

I would do: 

describe: `renders a Modal with correct elements when isLoading is false`
(I would probably query everything other then the first with `within` modal, but that's not really necessary)
-> before all with isLoading false
it - renders really a modal (just to be sure that it's the right component and element)
it - heading correct
it - content correct
it - close button enabled
it - approve button enabled
it - reject button enabled

describe: `renders a Modal with disabled interactions when isLoading is true`
-> before all with isLoading true
it - close button disabled
it - approve button disabled
it - reject button disabled

describe: 'handles user interaction'
-> before each 
it - user can close modal
it - user can approve
it - user can reject

That's more readable and if there is a failing test, it's easy to see which assertion is failing. 

It also makes sure the interactions are tested in isolation. Currently, if there would be a bug where clicking the close button would trigger closing and rejecting but the reject button does nothing, it still would be a green test.

Reply:
Ok I understand what you mean now. What you are saying was that having 48 assertions on different elements in a single `it` block is unpleasant. And that each `it` block should asserty a single element? And I see why now ^^

Thank you!

Reply:
[ce76e91](https://github.com/aiven/klaw/pull/611/commits/ce76e91b9d29800e09573ccd2a40936cb6743a68)